testing/yield.js

```diff
@@ -24,44 +26,47 @@ export function installYieldIt(realIt) {
   it.skip = realIt.skip;
 }
 
+/**
+ * A convenient method so you can flush the event queue by doing
+ * `yield macroTask()` in your test.
+ * @returns {Promise}
+ */
+export function macroTask() {
+  return new Promise(setTimeout);
+}
+
 function enableYield(fn, message, runnable) {
   if (!runnable || !runnable.constructor
       || runnable.constructor.name !== 'GeneratorFunction') {
     return fn(message, runnable);
   }
   return fn(message, done => {
-    const runner = (iterator, result) => {
+    const iterator = runnable();
+    function step(method, result) {
       let state;
       try {
-        state = iterator.next(result);
+        state = iterator[method](result);
       } catch (e) {
         // catch any assertion errors and pass to `done`
         // otherwise the messages are swallowed
         return done(e);
       }
       if (state.done) {
-        return done();
+        Promise.resolve(state.value).then(() => done(), _throw);
```

Comment:
This is incorrect, the iterator is done and pushing the error into it won't be handled. This should be `then(() => done(), done)`